app/controllers/api/v3/analytics_controller.rb

```diff
@@ -1,14 +1,18 @@
 class Api::V3::AnalyticsController < APIController
-  before_action :set_timezone_to_IST
-  after_action :set_timezone_to_UTC
+  around_action :set_time_zone
 
   private
 
-  def set_timezone_to_IST
-    Groupdate.time_zone = "Asia/Kolkata"
-  end
+  def set_time_zone
+    time_zone = Rails.application.config.country[:time_zone] || DEFAULT_ANALYTICS_TIME_ZONE
+
+    Groupdate.time_zone = time_zone
+
+    Time.use_zone(time_zone) do
```

Comment:
It is intentional. The reason why I've added this is because tz are more important now since dates are *also* being generated in JS. For a fair comparison of JS dates and dates generated server-side, it's valuable to generate server-side dates in the best guess timezone for the particular country where the progress report is being rendered.

Now of course, there's exceptions to this, like you could be a nurse from Punjab, entering patients in New York. But that should be rare and minimal.